### PR TITLE
Improve the list edit view by putting the extra actions in a popup menu

### PR DESCRIPTION
**Pull request by brunnerant**
See #170 for an explanation of what this PR does.

### CODE REVIEW

**Comment by brunnerant** [comment]
I have decided to also link the different activities together in this PR, because it was not such a big amount of work. It means that quizzes can now be answered from the quiz list, and treasure hunt quizzes are also dispatched correctly.

File: app/src/main/java/ch/epfl/qedit/backend/auth/MockAuthService.java [hunk 1]
```diff
@@ -34,10 +34,10 @@ public IdlingResource getIdlingResource() {
         return idlingResource;
     }
 
-    /** Simply make the current thread wait 2 second */
-    private static void wait2second() {
+    /** Simply make the current thread wait 0.5 seconds, to do as if the request takes time */
+    private static void fakeWait() {
         try {
-            Thread.sleep(2000);
+            Thread.sleep(500);
```
Comment (nbaehler) [inline app/src/main/java/ch/epfl/qedit/backend/auth/MockAuthService.java:40]: You do this to speed up our tests, am I right?

File: app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java [hunk 2]
```diff
@@ -79,61 +81,57 @@
...
-            }
+        // This is called when an item in the popup menu is clicked
+        public boolean onMenuItemClick(MenuItem item) {
+            int id = item.getItemId();
+
+            // We pass as an event the index of the item in the popup menu
+            adapter.notifyItem(getAdapterPosition(), id);
 
-            adapter.notifyItem(adapter.selectedQuestion, EventType.Select);
+            return true; // the even was handled
```
Comment (nbaehler) [inline app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java:134]: typo: event

File: app/src/main/java/ch/epfl/qedit/backend/auth/MockAuthService.java (hunk 1 above)
Comment (brunnerant) [reply app/src/main/java/ch/epfl/qedit/backend/auth/MockAuthService.java:40]: Yes, exactly. And also to speed up the demonstration that will be using the mocks.

File: app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java (hunk 2 above)
Comment (brunnerant) [reply app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java:134]: Thanks, I will will fix it.

**Comment by brunnerant** [comment]
Yeah, I felt like it would make more sense to put it there. And HomeQuizListFragment was starting to be pretty big, so it was also a good move.

**Pull request merged**